### ISSUE TITLE
App freezing with uwsgi

### ISSUE THREAD

**Issue by aidos** [segmentio/analytics-python#55]
I've just been trying to track down a really weird issue with running segment in flask under uwsgi. I'm explicitly creating the segment client myself.

The curious (non-app loading) bits of uwsgi look like this:

````
uwsgi --master -p 1 --vacuum --enable-threads \
        --carbon stats-collection.prod.service.rapidtender.com:2003 \
        --carbon-name-resolve
````

I actually contact segment during the starting of the app - which may be what's causing the issue. The code looks a little like this:

````
app = Flask(__name__)
app.analytics = analytics.Client(write_key='blah')
app.analytics.identify(user account info)
````

I have a whole lot of totally separate instances of the app and some of them load and some don't with no obvious reason for the differences. It won't even accept the first request.

If I add `app.analytics.flush()` after this bit of code - so it forces the thread to block and pump the data out to segment then the apps can start fine. I don't understand it at all but it seems to be something to do with running that thread before uwsgi takes control of the app. Or something.

I thought I'd post it here in case someone else runs into the same issue.

**Comment by aidos** [comment]
So I found the fix for this (in my case).

I enabled `--lazy-apps` in the uwsgi config as detailed in this stackoverflow question. It causes the the app init to happen *after* uwsgi has forked the workers. The trade off is less copy-on-write shared memory. In my case that's not an issue because I only run a single worker.

http://stackoverflow.com/questions/22752521/uwsgi-flask-sqlalchemy-and-postgres-ssl-error-decryption-failed-or-bad-reco
